src/main/webapp/assets/js/productpage.js

```diff
@@ -147,4 +147,20 @@ function createEditBusinessLink(isAdmin) {
   
   
   
+}
+
+function generateRating(rating){
+  if (rating == 404){
+    return "No ratings yet";
+  }
+  
+  var starHTML = '';
+  for (let i=0; i<5; i++) {
+    if (i+0.5<=rating) {
+      starHTML += '<i class="fas fa-star yellow-star"></i>';
+    } else {
+      starHTML += '<i class="fas fa-star"></i>';
+    }
+  }
+  return parseFloat(rating).toFixed(2) + " " + starHTML;
```

Comment:
Just wondering if you can process the rounding up in the backend instead? Cos in the home page, the rating will show 3.666667 instead of being rounded up.